src/coreclr/jit/codegenxarch.cpp

```diff
@@ -3178,14 +3163,24 @@ void CodeGen::genCodeForInitBlkUnroll(GenTreeBlk* node)
             regSize = min(regSize, YMM_REGSIZE_BYTES);
         }
 
-        bool zeroing = false;
         if (src->gtSkipReloadOrCopy()->IsIntegralConst(0))
         {
             // If the source is constant 0 then always use xorps, it's faster
             // than copying the constant from a GPR to a XMM register.
             emit->emitIns_SIMD_R_R_R(INS_xorps, EA_ATTR(regSize), srcXmmReg, srcXmmReg, srcXmmReg);
-            zeroing = true;
         }
+#ifdef FEATURE_SIMD
+        else if (src->gtSkipReloadOrCopy()->IsIntegralConst())
+        {
+            // Populate a constant vector from the fill value and save it
+            // to the data section so we can load it by address.
+            assert(regSize <= ZMM_REGSIZE_BYTES);
+            simd64_t constValue;
+            memset(&constValue, (uint8_t)(src->AsIntCon()->IconValue() & 0xFF), sizeof(simd64_t));
+            var_types loadType = compiler->getSIMDTypeForSize(regSize);
+            genSetRegToConst(srcXmmReg, loadType, compiler->gtNewVconNode(loadType, &constValue));
```

Comment:
There's some other places in codegen where we avoid doing `gtNew*Node` in favour of creating the type on the stack and passing its address. Is it worth doing the same here?

Reply:
What's the downside? it is simplified a large piece

![image](https://user-images.githubusercontent.com/523221/235920332-6cdad70c-066a-40fc-a2e6-59ce05fd3924.png)

Reply:
Sorry, might've worded this badly. I meant for example in `genCodeForReturnTrap` (as an example, we have many places that do this):
```cpp
    GenTreeIntCon cns = intForm(TYP_INT, 0);
    cns.SetContained();
    GetEmitter()->emitInsBinary(INS_cmp, emitTypeSize(TYP_INT), data, &cns);
```

So similarly, this could be:
```cpp
GenTreeVecCon vecCon = GenTreeVecCon(loadType);
memcpy(&vecCon->gtSimdVal, data, genTypeSize(type));
genSetRegToConst(srcXmmReg, loadType, &vecCon);
```

It avoids allocating the extra node for a very-short lived codegen only scenario. Alternatively there could just be a `genSetRegToConst` helper that takes in a `simd_t` (which we already define to represent the largest size for the target) and does the fast/efficient thing for us.

Reply:
> Alternatively there could just be a `genSetRegToConst` helper that takes in a simd_t (which we already define to represent the largest size for the target) and does the fast/efficient thing for us.

That would be best. `indirForm` and friends are hacks working around emitter interface deficiencies. It is not a pattern we want to see in new code.

Reply:
does it really makes sense to introduce 4 different `genSetRegToConst` for each simdX_t for this single use case? (that doesn't have a lot of hits judging by the diffs)

Reply:
addressed in https://github.com/dotnet/runtime/pull/85501/commits/13cf9e099cb9ca953b53439d4186a53322a6592d